internal/router/router.go

```diff
@@ -192,6 +192,17 @@ func (id *ResolvedURL) PreferredURL() *charm.URL {
 	return u
 }
 
+// DocPromulgatedURL the promulgated URL that
+// should be stored in the mongodoc.Entity.
+// It does not have the channel set.
+func (id *ResolvedURL) DocPromulgatedURL() *charm.URL {
```

Comment:
Not sure about this name, but I don;t have great alternatives. DocPromulgatedURL is like "use this only when you are working with mongo docs". Something like PromulgatedURLWithoutChannel would make the method reusable in other contexts. Take it or leave it.